### PR TITLE
router: include attempt count in virtual hosts

### PR DESCRIPTION
**Pull request by ramaraochavali**
Some services would like to know/log the request attempt (number of retry) being executed.  I think  it does not harm to add that for all virtual hosts. This PR adds https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/route/route_components.proto#envoy-api-field-route-virtualhost-include-request-attempt-count to virtual hosts
Fixes https://github.com/istio/istio/issues/10194
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by rshriram** [comment]
if this is new field, you need version guard

**Comment by ramaraochavali** [comment]
@rshriram This is not a new field and has been there for quite some time.

**Comment by hzxuzhonghu** [comment]
https://www.envoyproxy.io/docs/envoy/latest/intro/version_history#dec-20-2018 Not sure it is used by which version of istio

**Comment by howardjohn** [comment]
1.4 == Envoy 1.12
1.5 == Envoy 1.13

so it's been around for many releases

**Comment by ramaraochavali** [comment]
Also fixes this issue https://github.com/istio/istio/issues/10194

File: pilot/pkg/networking/core/v1alpha3/httproute.go [hunk 1]
```diff
@@ -88,9 +88,10 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundHTTPRouteConfig(
 	defaultRoute := istio_route.BuildDefaultHTTPInboundRoute(node, clusterName, traceOperation)
 
 	inboundVHost := &route.VirtualHost{
-		Name:    fmt.Sprintf("%s|http|%d", model.TrafficDirectionInbound, instance.ServicePort.Port),
-		Domains: []string{"*"},
-		Routes:  []*route.Route{defaultRoute},
+		Name:                       fmt.Sprintf("%s|http|%d", model.TrafficDirectionInbound, instance.ServicePort.Port),
+		Domains:                    []string{"*"},
+		Routes:                     []*route.Route{defaultRoute},
+		IncludeRequestAttemptCount: true,
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/httproute.go]: As retry happens at upstream, not sure if the inbound setting will override the incoming headers.

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/httproute.go]: That is a good point and also we do not need it for inbound routes. So removed it.

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/httproute.go]: ok

**Pull request merged**